### PR TITLE
Expose restaurants endpoint

### PR DESCRIPTION
**Pull request by fottander**
PT link: https://www.pivotaltracker.com/story/show/143404131

**Changes proposed in this PR:**
Adds test for viewing restaurants endpoints
Adds the controller and the route for the restaurants

**What I have learned working on this feature:**
We have learned how to make request specs. 
We have learned how to namespace api routes.

### CODE REVIEW

**Comment by tochman** [comment]
There are no pages in an API. I usually refer to `endpoints`

File: app/controllers/api/v1/list_restaurants_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Api::V1::ListRestaurantsController < ApplicationController
+  def index
+    restaurants = Restaurant.all
+    render json: {entries: restaurants}
```
Comment (AmberWilkie) [inline app/controllers/api/v1/list_restaurants_controller.rb]: Guys, you can actually view your endpoints by going to the URL (as long as it is a `get` request). See what this gives you. It is likely you want to format this response with `jbuilder`.

File: config/routes.rb [hunk 2]
```diff
@@ -21,5 +21,8 @@
     namespace :v0 do
       resources :ping, only: [:index], constraints: { format: 'json' }
     end
+    namespace :v1 do
+      resources :list_restaurants, only: [:index], constraints: { format: 'json' }
```
Comment (AmberWilkie) [inline config/routes.rb]: Nope. This should be `resources :restaurants`

File: spec/requests/api/v1/list_restaurants_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+RSpec.describe Api::V1::ListRestaurantsController, type: :request do
+  describe 'GET /v1/list_restaurants' do
+    it 'should return all the restaurants' do
+      create(:restaurant)
+      get '/api/v1/list_restaurants'
+      json_response = JSON.parse(response.body)
+      expect(response.status).to eq 200
+      expect(json_response['entries'].first['name']).to eq "Nisses"
+    end
```
Comment (AmberWilkie) [inline spec/requests/api/v1/list_restaurants_spec.rb]: This request spec should have an actual `expected_response` that lists a JSON response from the endpoint with one restaurant. You are deciding how you want to output your response, then making the code do that. How should this data be formatted to make it easy for the client / your front end team to consume it?

File: app/controllers/api/v1/list_restaurants_controller.rb (hunk 1 above)
Comment (tochman) [reply app/controllers/api/v1/list_restaurants_controller.rb]: `ListRestaurantsController` sets a narrow scope for this controller. I would suggest `RestaurantsController`

Comment (fottander) [reply app/controllers/api/v1/list_restaurants_controller.rb]: Alright so it´s ok to have RestaurantsController even if we have another one in the regular controllers?

File: app/views/api/v1/restaurants/index.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,6 @@
+json.entries do
+    json.array! @restaurants.each do |restaurant|
```
Comment (AmberWilkie) [inline app/views/api/v1/restaurants/index.json.jbuilder:2]: What's with the `!`?

File: config/routes.rb [hunk 5]
```diff
@@ -21,5 +21,8 @@
     namespace :v0 do
       resources :ping, only: [:index], constraints: { format: 'json' }
     end
```
Comment (AmberWilkie) [inline config/routes.rb]: This `v0` is unnecessary - it was for doing a very simple API endpoint in Cooper.

File: spec/requests/api/v1/restaurants_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+  describe 'GET /v1/restaurants' do
+    it 'should return all the restaurant´s name and description' do
+      restaurant = create(:restaurant)
+      get '/api/v1/restaurants'
+
+      expected_response = {
+        'entries' => [{
+          'name' => restaurant.name, 'description' => restaurant.description
+        }]     }
```
Comment (tochman) [inline spec/requests/api/v1/restaurants_spec.rb]: Please format this part.

File: spec/requests/api/v1/restaurants_spec.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::RestaurantsController, type: :request do
+
+  describe 'GET /v1/restaurants' do
+    it 'should return all the restaurant´s name and description' do
+      restaurant = create(:restaurant)
```
Comment (tochman) [inline spec/requests/api/v1/restaurants_spec.rb:7]: This is okay for now but I usually prefer to create my objects using `let` or `let!` 
```
let(:restaurant) { create(:restaurant) }

File: spec/support/response_json.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+module ResponseJSON
+  def response_json
+    JSON.parse(response.body)
+  end
+end
```
Comment (tochman) [inline spec/support/response_json.rb:5]: 👍🏼

**Review by tochman — CHANGES_REQUESTED** [review]
Small updates needed

File: config/routes.rb (hunk 5 above)
Comment (fottander) [reply config/routes.rb]: Yeah Thomas wanted us to do that before we started on the restaurants. So i guess we can delete those tests now.

File: app/views/api/v1/restaurants/index.json.jbuilder (hunk 4 above)
Comment (fottander) [reply app/views/api/v1/restaurants/index.json.jbuilder:2]: To create the correct endpoint output we needed this here, thomas helped us with this. Probably not an answer but something :)

**Review by tochman — APPROVED** [review]

**Pull request merged**